pkg/controllers/pushsecret/pushsecret_controller.go

```diff
@@ -72,9 +72,43 @@ type Reconciler struct {
 	ControllerClass string
 }
 
-func (r *Reconciler) SetupWithManager(mgr ctrl.Manager, opts controller.Options) error {
+func (r *Reconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager, opts controller.Options) error {
 	r.recorder = mgr.GetEventRecorderFor("pushsecret")
 
+	// Index PushSecrets by the stores they have pushed to (for finalizer management)
+	if err := mgr.GetFieldIndexer().IndexField(ctx, &esapi.PushSecret{}, "status.syncedPushSecrets", func(obj client.Object) []string {
```

Comment:
Field indexing choice: For status.syncedPushSecrets, you’re extracting only the store name (dropping the kind). This risks collisions if both SecretStore/foo and ClusterSecretStore/foo exist, right? Is that even fine?